options_test.go

```diff
@@ -1,40 +1,90 @@
 package envbuilder_test
 
 import (
+	"bytes"
 	"testing"
 
 	"github.com/coder/envbuilder"
 	"github.com/coder/serpent"
 	"github.com/stretchr/testify/require"
 )
 
-type testEnvCase struct {
-	Env      string
-	EnvValue string
-	Expected interface{}
-	Actual   func() interface{}
-}
+// TestEnvOptionParsing tests that given environment variables of different types are handled as expected.
+func TestTestEnvOptionParsing(t *testing.T) {
+	t.Run("string", func(t *testing.T) {
+		t.Setenv("SETUP_SCRIPT", "setup.sh")
+		t.Setenv("INIT_COMMAND", "sleep infinity")
+
+		var o envbuilder.Options
+		err := runCLI(&o)
+		require.NoError(t, err)
+
+		require.Equal(t, o.SetupScript, "setup.sh")
+		require.Equal(t, o.InitCommand, "sleep infinity")
+	})
```

Comment:
I think this expresses the intent more concisely, but to be clear your test as it stands is perfectly fine as it is.

```suggestion
	t.Run("string", func(t *testing.T) {
	        const val = "setup.sh"
		t.Setenv("SETUP_SCRIPT", val)

		var o envbuilder.Options
		err := runCLI(&o)
		require.NoError(t, err)

		require.Equal(t, o.SetupScript, val)
	})
```

Reply:
I like this way but when having to test 6 options, it looks too much IMO. Wdyt?

Reply:
It would looks like:

```go
const skipRebuildValue = true
const gitCloneSingleBranchValue = false
const exitOnBuildFailureValue = false
// and so on...

```

Reply:
Oh no I just meant for this particular test; don't worry about applying it to the others.